### ISSUE TITLE
Task: move `Window` from `qt_main_window`, create obvious window protocol API

### ISSUE THREAD

**Issue by tlambert03** [napari/napari#4984]
## 🧰 Task
This is a very old task that has probably come up in many conversations, but I can't find a super relevant issue, so creating a new one.

The `qt_main_window.Window` class is probably one of our biggest leaks of `Model/View` logic.  Its the main public facing interface (at `viewer.window`) that still only resides in the `_qt` folder.  Moreover, the relationship between `qt_main_window.Window` and `qt_main_window._QtMainWindow` has always been a little amorphous...

The [window.py](https://github.com/napari/napari/blob/main/napari/window.py) is a file we've had for a long time that could serve as place to define a protocol that a backend would need to implement to provide a view onto the ViewerModel (which is essentially what `Window` is)...

I'm not immediately sure, but it seems like we would then likely only have _one_ class in the `qt_main_window` file, that provides the backend implementation of the `Window` protocol.

It would be an opportunity to clarify exactly what objects should be getting passed around (i.e. should we be passing the `Viewer` to the `Window` which then passes itself to the `_QtMainWindow`?  Or was that just the situation we ended up in)

This would also be relevant in the context of deprecating public access to `QtViewer`, since the Window is where any new public methods should be going

**Comment by Czaki** [comment]
> This would also be relevant in the context of deprecating public access to `QtViewer`, since the Window is where any new public methods should be going

Do you mean access to `.qt_viewer`, not to the class?

**Comment by tlambert03** [comment]
> Do you mean access to .qt_viewer, not to the class?

yep.  not referring to the fact that you use QtViewer directly, and that it is exported in `napari.qt`  (while that too puts us in a more difficult position in terms of development, it's not causing any particular problems and is unrelated to what I'm talking about here)

**Comment by Czaki** [comment]
> while that too puts us in a more difficult position in terms of development

but allow embedding viewer without touching QMainWindow and is a good starting point for minimal viewer.
